### ISSUE TITLE
Header file is not detecting 

### ISSUE THREAD

**Issue by Hax4us** [termux/termux-packages#2172]
I m facing a problem while Travis build
In short , error is about pcap header file
pcap.h not found

Here is my build.sh
https://github.com/Hax4us/termux-packages/blob/hping/packages/hping3/build.sh

**Comment by Wetitpig** [comment]
You can try to remove that line and see what happens. Try changing the code until you can get it built. Then patch all files as diff.

**Comment by Hax4us** [comment]
@Wetitpig but I can't figured out the root cause and I compiled successfully in my system but I don't know why Travis is giving error of pcap header file

**Comment by Neo-Oli** [comment]
> You could perhaps run sudo apt-get install -y libpcap-dev in the pre-configure step as a hack-y work around.

No. This is what this file is for: https://github.com/termux/termux-packages/blob/master/scripts/setup-ubuntu.sh

**Comment by Grimler91** [comment]
@Neo-Oli I might be mistaken, but doesn't travis just download the docker image and start the build (without running setup-ubuntu.sh)? Adding libpcap-dev to setup-ubuntu.sh is definitely necessary when opening a pull request for hping3 though.

**Comment by Hax4us** [comment]
so i will have to wait for adding libpcap-dev in ubuntu.sh Script ? @Grimler91

**Comment by Neo-Oli** [comment]
@Grimler91 I think the Docker file explicitly calls setup-ubuntu.sh here:   https://github.com/termux/termux-packages/blob/4842499272aa814303244bff5e6606a1f7a61c68/scripts/Dockerfile#L23

**Comment by Grimler91** [comment]
I think that file is only used when Fornwall updates the docker image. 

Easy to test this though. @Hax4us try adding libpcap-dev to setup-ubuntu.sh as @Neo-Oli suggested, if that doesn't work you will need to do the hack-y `sudo apt-get install -y libpcap-dev`.

libpcap-dev won't be included in the docker image before any packages need it (to keep the image as small as possible).

**Comment by Neo-Oli** [comment]
@Grimler91 It probably needs both then. If a package requires it it should go into the setup-ubuntu.sh file, but to test it before the docker file is updated you'd need to do the hacky thing.

**Comment by Grimler91** [comment]
> If a package requires it it should go into the setup-ubuntu.sh file, but to test it before the docker file is updated you'd need to do the hacky thing.

Yeah I think you're right.

**Comment by Hax4us** [comment]
So guys as all of you suggested i tried but now the travis is giving error like that
./build-package.sh: line 948: QUIET_BUILD: unbound variable

Here is full log
https://api.travis-ci.org/v3/job/343741170/log.txt

**Comment by Grimler91** [comment]
That's fixed in this pending pull request: https://github.com/termux/termux-packages/pull/2171/files

**Comment by Hax4us** [comment]
any workaround for now ? Well i think it will take long time to merge by fornwall so i will have to wait :(

**Comment by Grimler91** [comment]
@Hax4us You can apply the same changes in your branch/merge that pull request onto your branch.

**Comment by tomty89** [comment]
Don't packages in `TERMUX_PKG_DEPENDS` (and their -dev) get installed automatically?

**Comment by Neo-Oli** [comment]
@tomty89 `TERMUX_PKG_DEPENDS` is for things to be installed on Termux. This issue is about the package-builder machine needing an additional program to create a package. 

That said, looking at this more closely it's weird that the builder machine would require a `-dev` package. It should probably not do that at all, since it might be a different version than the termux version. The program should link against the built `pcap.h` for termux, rather than the host machine's one. It's probably the makefile that needs patching instead.

**Comment by Hax4us** [comment]
Here is makefile content

------------------------------------------------

# $smu-mark$ 
# $name: Makefile.in$ 
# $author: Salvatore Sanfilippo 'antirez'$ 
# $copyright: Copyright (C) 1999 by Salvatore Sanfilippo$ 
# $license: This software is under GPL version 2 of license$ 
# $date: Sun Jul 25 17:56:15 MET DST 1999$ 
# $rev: 3$ 

CC= gcc
AR=/data/data/com.termux/files/usr/bin/ar
RANLIB=/data/data/com.termux/files/usr/bin/ranlib
CCOPT= -O2 -Wall   -DUSE_TCL
DEBUG= -g
#uncomment the following if you need libpcap based build under linux
#(not raccomanded)
COMPILE_TIME=
INSTALL_MANPATH=/data/data/com.termux/files/usr/share/man
PCAP=-lpcap

ARSOBJ = ars.o apd.o split.o rapd.o

OBJ=	main.o getifname.o getlhs.o \
	parseoptions.o datafiller.o \
	datahandler.o gethostname.o \
	binding.o getusec.o opensockraw.o \
	logicmp.o waitpacket.o resolve.o \
	sendip.o sendicmp.o sendudp.o \
	sendtcp.o cksum.o statistics.o \
	usage.o version.o antigetopt.o \
	sockopt.o listen.o \
	sendhcmp.o memstr.o rtt.o \
	relid.o sendip_handler.o \
	libpcap_stuff.o memlockall.o memunlockall.o \
	memlock.o memunlock.o ip_opt_build.o \
	display_ipopt.o sendrawip.o signal.o send.o \
	strlcpy.o arsglue.o random.o scan.o \
	hstring.o script.o interface.o \
	adbuf.o hex.o apdutils.o sbignum.o \
	sbignum-tables.o $(ARSOBJ)

all: .depend hping3

dep: .depend
.depend:
	@echo Making dependences
	@$(CC) -MM *.c > .depend

libars.a: $(ARSOBJ)
	$(AR) rc $@ $^
	$(RANLIB) $@

hping3: byteorder.h $(OBJ)
	$(CC) -o hping3 $(CCOPT) $(DEBUG) $(OBJ) -L/data/data/com.termux/files/usr/lib $(PCAP)  -ltcl8.6 -lm -lpthread -landroid-shmem
	@echo
	./hping3 -v
	@echo "use \`make strip' to strip hping3 binary"
	@echo "use \`make install' to install hping3"

hping3-static: byteorder.h $(OBJ)
	$(CC) -static -o hping3-static $(CCOPT) $(DEBUG) $(OBJ) -L/data/data/com.termux/files/usr/lib $(PCAP)  -ltcl8.6 -lm -lpthread -ldl

byteorder.h:
	./configure

.c.o:
	$(CC) -c $(CCOPT) $(DEBUG) $(COMPILE_TIME) $<

clean:
	rm -rf hping3 *.o libars.a

distclean:
	rm -rf hping3 *.o byteorder byteorder.h systype.h Makefile libars.a .depend

install: hping3
	cp -f hping3 /data/data/com.termux/files/usr/bin/
	chmod 755 /data/data/com.termux/files/usr/bin/hping3
	ln -s /data/data/com.termux/files/usr/bin/hping3 /data/data/com.termux/files/usr/bin/hping
	ln -s /data/data/com.termux/files/usr/bin/hping3 /data/data/com.termux/files/usr/bin/hping2
	@if [ -d ${INSTALL_MANPATH}/man8 ]; then \
		cp ./docs/hping3.8 ${INSTALL_MANPATH}/man8; \
		chmod 644 ${INSTALL_MANPATH}/man8/hping3.8; \
	else \
		echo "@@@@@@ WARNING @@@@@@"; \
		echo "Can't install the man page: ${INSTALL_MANPATH}/man8 does not exist"; \
	fi

strip: hping3
	@ls -l ./hping3
	strip hping3
	@ls -l ./hping3

include .depend

**Comment by tomty89** [comment]
@Neo-Oli 
> The program should link against the built pcap.h for termux
So packages in `TERMUX_PKG_DEPENDS` do get built automatically before the specific package itself gets built right? (And the Makefile should look for headers and and libraries of the builds)
> it's weird that the builder machine would require a -dev package
That's where I got confused, wondering how the build script deal with build time dependency and why certain packages exist in the setup script.

**Comment by Neo-Oli** [comment]
@tomty89 Yes. The setup-ubuntu.sh file contains programs that need to be installed in order for the build of a porgram to work. Programs like `make` and `pkg-config`. Sometimes a program requires a weird build tool and that's when it gets added there. But header files to link against generally are built for termux and the linker _usually_ picks them up but sometimes the linker needs to be specifically told where the files are. https://github.com/termux/termux-packages/blob/698b15615cbcee71e23de59014143d48ab3d2c97/disabled-packages/libwebp/build.sh#L18

**Comment by Neo-Oli** [comment]
@Hax4us It looks like your makefile doesn't use the `$CFLAGS` and `$CPPFLAGS` that the build scripts provides. These mostly contain the cross-compiling options, but also tell the linker where to pick up includes.
https://github.com/termux/termux-packages/blob/4842499272aa814303244bff5e6606a1f7a61c68/build-package.sh#L592
You need to patch the Makefile to add these options. A few programs which use non-standard Makefiles require this. https://github.com/termux/termux-packages/blob/4842499272aa814303244bff5e6606a1f7a61c68/packages/sl/makefile.patch#L13

**Comment by Hax4us** [comment]
@Neo-Oli thnx u were right ....now pcap.h error Solved :)
